src/ucp/rndv/proto_rndv.h

```diff
@@ -68,6 +68,9 @@ typedef struct {
  */
 typedef struct {
     ucp_proto_rndv_ack_priv_t super;
+
+    /* Memory type of fragment buffers which are used by get/mtype and put/mtype
+     * protocols */
```

Comment:
maybe need another priv struct for get/mtype and put/mtype, since zcopy protocols do not need it?

Reply:
imo, code will look uglier. Need to remove `ucp_proto_select_add_proto` from `ucp_proto_rndv_get_common_probe` and `ucp_proto_rndv_put_common_probe` which increases code duplication. Or pass more parameters, like `priv`, `priv_size` and `priv_offset_to_bulk` to these `common_probe` routines